esi_ui/static/dashboard/project/esi/nodes/esi-nodes-table.controller.js

```diff
@@ -206,6 +208,60 @@
         }
       });
     }
+
+    function manageFloatingIPs(node) {
+      var launchContext = {
+        ports: [],
+        detachable_floating_ips: [],
+      };
+
+      var network_port_names = node['network_port_names'];
+      var floating_ips = node['floating_ips'];
+      for (var i = 0; i < floating_ips.length; i++) {
+        if (network_port_names[i].length > 0) {
+          if (floating_ips[i].length == 0) {
+            launchContext.ports.push(network_port_names[i]);
+          }
+          if (floating_ips[i].length > 0) {
+            launchContext.detachable_floating_ips.push(floating_ips[i] + ' (' + network_port_names[i] + ')');
+          }
+        }
+      }
+
+      manageFloatingIPsModalService.open(launchContext)
+      .then(function (response) {
+        const stepModels = response.stepModels;
+        const floatingIPs = response.floatingIPs;
+        const ports = response.ports;
+        if (stepModels.action === 'attach') {
+          spinnerService.showModalSpinner('Attaching Floating IP');
+          const floatingIPAddress = stepModels.attach.floating_ip.floating_ip_address;
+          const floatingIPId = floatingIPAddress === 'Create new one' ? 'Create new one' : floatingIPs.filter(ip => ip.ip === floatingIPAddress)[0]?.id;
```

Comment:
We probably don't need this 'Create new one' logic since we are only attaching/detaching, and not creating.

Reply:
I think it's worth having; making the user go to the Compute section to create a floating IP is a pain. This feels like a better streamlined experience.